packages/table/src/table2.tsx

```diff
@@ -1478,30 +1470,21 @@ export class Table2 extends AbstractComponent<Table2Props, TableState, TableSnap
     private syncViewportPosition = ({ nextScrollLeft, nextScrollTop }: TableSnapshot) => {
         const { viewportRect } = this.state;
 
-        if (this.scrollContainerElement == null || this.columnHeaderElement == null || viewportRect === undefined) {
+        if (this.scrollContainerElement == null || viewportRect === undefined) {
             return;
         }
 
         if (nextScrollLeft !== undefined || nextScrollTop !== undefined) {
-            // we need to modify the scroll container explicitly for the viewport to shift. in so
-            // doing, we add the size of the header elements, which are not technically part of the
-            // "grid" concept (the grid only consists of body cells at present).
```

Comment:
This is not true. `scrollTop` should not be dependent on header size here:


https://github.com/palantir/blueprint/assets/40673216/e06954c5-f750-4029-a020-f45745a988d8

Reply:
if that's the case then I'd expect the viewportRect to not include space for the headers, re https://github.com/palantir/blueprint/pull/6882/files#diff-c2ca442c1f8367c203de64884a8526876c32365d9dd5f071887fb16183ada632R100

Reply:
> if that's the case then I'd expect the viewportRect to not include space for the headers

I don't think we can overload the semantics of `viewportRect`. `viewportRect` is interpreted as `Table2`'s rect in other codepaths – for example, here:

https://github.com/palantir/blueprint/blob/f2e9d6e049b3c52eef4ea7d9cb080b4dca23c7a1/packages/table/src/locator.ts#L185-L198


That being said, I do think there is space for refactors. I consider that outside of the scope for this PR though.

Reply:
`viewportRect` does include space for the headers (it contains headers, frozen cells, and the visible "scrollable cells"). The absolute placement of the scrollable cells within the scrollable element `S` is such that `S.scrollTop` need not take frozen cell or header height into account. Hopefully the diagram in https://github.com/palantir/blueprint/pull/6882#issuecomment-2257268989 helps explain this a bit better.